### PR TITLE
[8.0] Add module base_user_auth_log

### PR DESCRIPTION
**Pull request by alexis-via**
See README.rst for more info.

### CODE REVIEW

**Review by cdubuit — APPROVED** [review]
Test, no code review.
Great idea.

**Comment by pedrobaeza** [comment]
https://github.com/OCA/server-tools/tree/8.0/auth_brute_force also logs logins, so I think this one is overlapping functionality.

**Review by legalsylvain — CHANGES_REQUESTED** [review]
Hi @alexis-via, thanks for your contribution.

1. @pedrobaeza +1. 

@alexis-via, what did you not found in auth_brute_force to create this module. It could be better to improve auth_brute_force, instead of creating two ways to log authentication.
In a nutshell, auth_brute_force : 
- log authentications (as your module)
- ban users (by IP) that tried too many times to authenticate, with bad logins. (existing or not)

2. Tested on the last runbot instance, your module create 3 logs per try. Do you have an idea why ? 

image

kind regards.

**Comment by legalsylvain** [comment]
3. Are you sure ```default=datetime.utcnow()``` is working correctly ? all the logs have the same authentication date, that is wrong.

File: base_user_auth_log/models/res_users_auth_log.py [hunk 1]
```diff
@@ -17,8 +16,7 @@ class ResUsersAuthLog(models.Model):
         'res.users', string='User', ondelete='cascade', readonly=True)
     login = fields.Char(string='Login', readonly=True)
     date = fields.Datetime(
-        string='Authentication Date', required=True, readonly=True,
-        default=datetime.utcnow())
```
Comment (sbidoul) [inline base_user_auth_log/models/res_users_auth_log.py]: `default=datetime.utcnow`, without `()` should work.

**Comment by dreispt** [comment]
Should this be closed in favour of `auth_brute_force`?

**Comment by legalsylvain** [comment]
Closing. (Duplicate with auth brut force)
One idea of improvment could be to make a PR with one module to make logs, and another one auth brut force, depending of the first one, and adding brut force features.
(Need refactoring)

**Pull request closed without merging**